fsw/cfe-core/unit-test/es_UT.c

```diff
@@ -4797,10 +4801,17 @@ void TestAPI(void)
     CFE_ES_Global.AppTable[0].AppState = CFE_ES_AppState_RUNNING;
     CFE_ES_Global.AppTable[0].TaskInfo.MainTaskId = 15;
     OS_TaskCreate(&CFE_ES_Global.TaskTable[1].TaskId, NULL, NULL, NULL,
-                  0, 0, 0);
+                  0, 0, 0);    
+    DeleteChildResult = CFE_ES_DeleteChildTask(CFE_ES_Global.TaskTable[1].TaskId);    
+    CFE_ES_GetAppInfo(&AppInfo,CFE_ES_Global.TaskTable[1].AppId);    
+
+    if( DeleteChildResult == CFE_SUCCESS && CFE_ES_Global.RegisteredTasks == 15
+        && AppInfo.NumOfChildTasks == 0 )
+    {
+       DeleteChildTestResult = CFE_SUCCESS;
+    }
```

Comment:
Each of these conditions should be a separate assert.  When combined into a single assert then it is not possible to see what went wrong from a log file, making it difficult to diagnose.

Please use the `UtAssert_True()` macro to test each individual condition you are looking for, and also output the actual values too (not just a generic text) for debug.

The general pattern should be something like this:

```
Return = CFE_ES_DeleteChildTask(CFE_ES_Global.TaskTable[1].TaskId);
UtAssert_True(Return == CFE_SUCCESS, 
         "DeleteChildResult() return=%x", (unsigned int)Return);
Return = CFE_ES_GetAppInfo(&AppInfo,CFE_ES_Global.TaskTable[1].AppId);
UtAssert_True(Return == CFE_SUCCESS, 
         "CFE_ES_GetAppInfo() return=%x", (unsigned int)Return);
UtAssert_True(AppInfo.NumOfChildTasks == 0, 
         "AppInfo.NumOfChildTaskss == %u", (unsigned int)AppInfo.NumOfChildTasks);
```
And so on....

It also is not clear to me why we verify `CFE_ES_Global.RegisteredTasks == 15` as part of this test case.  I expected to see a call to CFE_ES_GetAppInfo before and after the call to CFE_ES_DeleteChildTask, to verify that the NumOfChildTasks output was the expected number.

Reply:
Also Note - asserting on each condition separately should also remove the need for extra local variables.